dnas/mewsfeed/zomes/coordinator/follows/src/follower_to_creators.rs

```diff
@@ -76,63 +117,67 @@ pub fn count_followers_for_creator(creator: AgentPubKey) -> ExternResult<usize>
     count_links(query)
 }
 
-#[hdk_extern]
-pub fn get_follower_links_for_creator(
-    input: GetFollowersForCreatorInput,
-) -> ExternResult<Vec<Link>> {
-    let mut links = get_links(input.creator, LinkTypes::CreatorToFollowers, None)?;
-    links.dedup_by_key(|l| l.target.clone());
-    let links_page = paginate_by_agentpubkey(links, input.page)?;
-
-    Ok(links_page)
-}
-
-#[hdk_extern]
-pub fn get_follower_link_details_for_creator(creator: AgentPubKey) -> ExternResult<LinkDetails> {
-    let links = get_link_details(creator, LinkTypes::CreatorToFollowers, None)?;
-
-    Ok(links)
-}
-
 #[hdk_extern]
 pub fn remove_creator_for_follower(input: RemoveCreatorForFollowerInput) -> ExternResult<()> {
-    let links = get_links(
-        input.base_follower.clone(),
-        LinkTypes::FollowerToCreators,
-        None,
-    )?;
-
-    for link in links {
-        let entry_hash =
-            EntryHash::try_from(link.target.clone()).map_err(|err| wasm_error!(err))?;
-        if AgentPubKey::from(entry_hash).eq(&input.target_creator) {
-            delete_link(link.create_link_hash)?;
-        }
-    }
-
-    let links = get_links(
-        input.target_creator.clone(),
-        LinkTypes::CreatorToFollowers,
-        None,
+    let _deleted_link_count: DeleteReport = call_local_zome(
+        "trust_atom",
+        "delete_trust_atoms",
+        AnyLinkableHash::from(input.target_creator),
     )?;
 
-    for link in links {
-        let entry_hash =
-            EntryHash::try_from(link.target.clone()).map_err(|err| wasm_error!(err))?;
-        if AgentPubKey::from(entry_hash).eq(&input.base_follower) {
-            delete_link(link.create_link_hash)?;
-        }
-    }
+    // let links = get_links(
+    //     input.base_follower.clone(),
+    //     LinkTypes::FollowerToCreators,
+    //     None,
+    // )?;
+
+    // for link in links {
+    //     if AgentPubKey::from(EntryHash::from(link.target.clone())).eq(&input.target_creator) {
+    //         delete_link(link.create_link_hash)?;
+    //     }
+    // }
+
+    // let links = get_links(
+    //     input.target_creator.clone(),
+    //     // TODO trust atoms instead
+    //     // LinkTypes::TrustAtom,
+    //     None,
+    // )?;
+
+    // for link in links {
+    //     if AgentPubKey::from(EntryHash::from(link.target.clone())).eq(&input.base_follower) {
+    //         delete_link(link.create_link_hash)?;
+    //     }
+    // }
 
     Ok(())
 }
 
 #[hdk_extern]
-pub fn follow(agent: AgentPubKey) -> ExternResult<()> {
+pub fn follow(input: FollowInput) -> ExternResult<()> {
```

Comment:
maybe all this logic should just be included in `add_creator_for_follower()`, and we can keep `follow()` as just an alias to it